enJsonCounter.py

```diff
@@ -13,195 +14,185 @@ def main():
-            for ele in consumedData :
-                if key == ele.text :
-                    if key in finalDict :
-                        finalDict[key].append(ele.key)
-                    else :
-                        finalDict[key] = [ele.key]
-
-        file.write("\n\n")
-        file.write("The following string values are duplicated. Underneath the value are the keys it is found in: \n")
-
-        #This is just printing items that are duplicated (>=2)
-        for ele in finalDict :
-            if (len(finalDict[ele]) >= 2) :
-                file.write("   " + ele + " : " + str(len(finalDict[ele])) + "\n")
-                for ele2 in finalDict[ele]:
-                    file.write("      " + ele2 + "\n")
+        # get this in a function
+        currentDateTime = str(datetime.datetime.now())
 
-        # use string builder?
-        file.write("\n")
-        file.write("The top 5 duplicated string are: ")
+        # TODO add report name (made it from cmd args)
+        f.write("Statistics Report for \n\n")
+        f.write("This report was ran on: " + currentDateTime + "\n")
+        # TODO get last modified date
+        f.write("The json file read was last updated on: \n")
 
+        #JsonData = json.load(open('exampleJSON.json', encoding='utf-8'))
+        #JsonData = json.load(open('exampleJSON.json'))
 
+        consumedData = []
 
-    #just going to return files for now and not worry about doing the totalling here as of now
-    def recursion(filepath, key, ArrayOfFiles, ArrayOfOccurences):
+        def bubbleSortByLength(listBoi):
 
-        global someLongVariableToTrackOccurences
+            arr = copy.deepcopy(listBoi)
 
-        if(filepath.is_dir()) :
-            for item in filepath.iterdir():
-                recursion(Path(item), key, ArrayOfFiles, ArrayOfOccurences)
-        else:
-            openItem = open(filepath, "r")
-            readItem = openItem.read()
-            occurences = re.findall(key, readItem)
-            ArrayOfFiles.append(filepath)
-            ArrayOfOccurences.append(len((occurences)))
+            n = len(arr)
 
-    def step3():
+            # Traverse through all array elements
+            for i in range(n):
 
-        ArrayOfFiles = []
+                # Last i elements are already in place
+                for j in range(0, n-i-1):
+
+                    # traverse the array from 0 to n-i-1
+                    # Swap if the element found is greater
+                    # than the next element
+                    if arr[j].wordCount < arr[j+1].wordCount:
+                        arr[j], arr[j+1] = arr[j+1], arr[j]
 
-        wordsList = []
-        total = 0
+                    return arr
+
+            def step1and2func():
 
-        start_time = time.time()
+                print("running report")
+                wordsList = []
+                total = 0
+                JsonData = json.load(open('exampleJSON.json', encoding='utf-8'))
```

Comment:
JsonData instantiation never got uncommented and so this doesn't exist and is giving me squiglies